build/secondary/third_party/googletest/BUILD.gn

```diff
@@ -1,266 +1,13 @@
-  testonly = true
-  sources = [ "googletest/test/gtest-unittest-api_test.cc" ]
-  deps = [ ":gtest" ]
-}
-
-group("gtest_all_tests") {
-  testonly = true
-  deps = [
-    ":gtest_all_test",
-    ":gtest_environment_test",
-    ":gtest_listener_test",
-    ":gtest_no_test",
-    ":gtest_param_test",
-    ":gtest_premature_exit_test",
-    ":gtest_repeat_test",
-    ":gtest_sole_header_test",
-    ":gtest_stress_test",
-    ":gtest_unittest_api_test",
-  ]
-}
-
-config("gmock_private_config") {
-  visibility = [ ":*" ]
-  include_dirs = [ "googlemock" ]
-}
-
-config("gmock_config") {
-  include_dirs = [ "googlemock/include" ]
-
-  cflags_cc = [
-    # The MOCK_METHODn() macros do not specify "override", which triggers this
-    # warning in users: "error: 'Method' overrides a member function but is not
-    # marked 'override' [-Werror,-Winconsistent-missing-override]". Suppress
-    # these warnings until https://github.com/google/googletest/issues/533 is
-    # fixed.
-    "-Wno-inconsistent-missing-override",
-  ]
-}
-
-static_library("gmock") {
-  testonly = true
-  public = [ "googlemock/include/gmock/gmock.h" ]
-  sources = [ "googlemock/src/gmock-all.cc" ]
-  public_configs = [ ":gmock_config" ]
-  configs += [ ":gmock_private_config" ]
-  deps = [ ":gtest" ]
-}
-
-static_library("gmock_main") {
-  testonly = true
-  sources = [ "googlemock/src/gmock_main.cc" ]
-  public_deps = [
-    ":gmock",
-    ":gtest",
-  ]
-}
-
-executable("gmock_all_test") {
-  testonly = true
-  sources = [
-    "googlemock/test/gmock-actions_test.cc",
-    "googlemock/test/gmock-cardinalities_test.cc",
-    "googlemock/test/gmock-generated-actions_test.cc",
-    "googlemock/test/gmock-generated-function-mockers_test.cc",
-    "googlemock/test/gmock-generated-internal-utils_test.cc",
-    "googlemock/test/gmock-generated-matchers_test.cc",
-    "googlemock/test/gmock-internal-utils_test.cc",
-    "googlemock/test/gmock-matchers_test.cc",
-    "googlemock/test/gmock-more-actions_test.cc",
-    "googlemock/test/gmock-nice-strict_test.cc",
-    "googlemock/test/gmock-port_test.cc",
-    "googlemock/test/gmock-spec-builders_test.cc",
-    "googlemock/test/gmock_test.cc",
-  ]
-  configs += [
-    ":gmock_private_config",
-    ":gtest_private_config",
-  ]
-  deps = [
-    ":gmock",
-    ":gmock_main",
-    ":gtest",
-  ]
-}
-
-executable("gmock_link_test") {
-  testonly = true
-  sources = [
-    "googlemock/test/gmock_link2_test.cc",
-    "googlemock/test/gmock_link_test.cc",
-    "googlemock/test/gmock_link_test.h",
-  ]
-  configs += [ ":gmock_private_config" ]
-  deps = [
-    ":gmock",
-    ":gmock_main",
-    ":gtest",
-  ]
-}
-
-executable("gmock_stress_test") {
+group("gmock") {
   testonly = true
-  sources = [ "googlemock/test/gmock_stress_test.cc" ]
-  configs += [ ":gmock_private_config" ]
-  deps = [
-    ":gmock",
-    ":gtest",
-  ]
+  public_deps = [ "//flutter/third_party/googletest:gmock" ]
 }
 
-group("gmock_all_tests") {
+group("gtest") {
```

Comment:
Why is this alias still necessary?

Reply:
This target is referenced by some other third_party deps as `//third_party/googletest:gtest`.

Reply:
Do we care to build the test targets (since they depend on gtest) of third party deps? Not blocking this.

Reply:
I don't understand how GN is working in this case, but as far as I can tell the flag here is false:

https://flutter.googlesource.com/third_party/abseil-cpp/+/refs/heads/main/BUILD.gn#194,

but GN goes looking for the dependencies of all of those targets anyway. We aren't building them, though.